crates/kornia-io/src/stream/capture.rs

```diff
@@ -102,13 +160,32 @@ impl StreamCapture {
     /// # Returns
     ///
     /// An Option containing the last captured Image or None if no image has been captured yet.
-    pub fn grab(&self) -> Result<Option<Image<u8, 3>>, StreamCaptureError> {
+    pub fn grab(&mut self) -> Result<Option<Image<u8, 3>>, StreamCaptureError> {
         if !self.running {
             return Err(StreamCaptureError::PipelineNotRunning);
         }
 
-        // SAFETY: we have a lock on the last_frame
-        Ok(self.last_frame.lock().unwrap().take())
+        // SAFETY: we have a lock on the buffer_pool
+        let mut buffer_pool = self.buffer_pool.lock().unwrap();
+        let Some(frame_buffer) = buffer_pool.dequeue() else {
+            return Ok(None);
+        };
+
+        let frame_buffer = std::mem::ManuallyDrop::new(frame_buffer);
```

Comment:
@gbin having a bit of trouble by trying to prevent gstreamer to deallocate the buffer. Do you have any insights here ?